core/src/androidMain/kotlin/io/tolgee/storage/TolgeeStorageProviderAndroid.kt

```diff
@@ -0,0 +1,35 @@
+package io.tolgee.storage
+
+import android.content.Context
+import java.io.File
+
+class TolgeeStorageProviderAndroid(
+  private val context: Context,
+  private val versionCode: Int,
+  private val path: String = "tolgee/localization-cache",
+) : TolgeeStorageProvider {
+  private val cacheDir get() = context.filesDir / path / versionCode.toString()
+
+  override fun put(name: String, data: ByteArray) {
+    val dir = cacheDir
+    dir.mkdirs()
+    val file = dir / escape(name)
+    file.writeBytes(data)
+  }
+
+  override fun get(name: String): ByteArray? {
+    val file = cacheDir / escape(name)
+    if (!file.exists()) {
+      return null
+    }
+    return file.readBytes()
```

Comment:
Should be wrapped like this:

```kotlin
return scopeCatching {
    file.readBytes()
}.getOrNull()
```

I/O operations cause unpredictable problems even you think it won't.  
So it's better to wrap it like this.